lib/core/presentation/pages/chat/chat_message/widgets/chat_settings_popup_menu.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:async';
+
+import 'package:adaptive_dialog/adaptive_dialog.dart';
+import 'package:app/core/presentation/widgets/floating_frosted_glass_dropdown_widget.dart';
+import 'package:app/core/presentation/widgets/future_loading_dialog.dart';
+import 'package:app/core/service/matrix/matrix_service.dart';
+import 'package:app/gen/assets.gen.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:matrix/matrix.dart';
+import 'package:app/core/presentation/widgets/theme_svg_icon_widget.dart';
+import 'package:app/core/presentation/dpos/common/dropdown_item_dpo.dart';
+
+enum ChatOptions { chatDetails, leave }
+
+class ChatSettingsPopupMenu extends StatefulWidget {
+  final Room room;
+
+  const ChatSettingsPopupMenu({Key? key, required this.room}) : super(key: key);
+
+  @override
+  ChatSettingsPopupMenuState createState() => ChatSettingsPopupMenuState();
+}
+
+class ChatSettingsPopupMenuState extends State<ChatSettingsPopupMenu> {
+  StreamSubscription? notificationChangeSub;
+
+  @override
+  void dispose() {
+    notificationChangeSub?.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final t = Translations.of(context);
+    notificationChangeSub ??= getIt<MatrixService>()
+        .client
```

Comment:
Since this is a stateful widget, put notificationChangeSub into `initState` instead